lnwallet/chancloser/chancloser.go

```diff
@@ -70,6 +71,11 @@ const (
 	// phase.
 	closeShutdownInitiated
 
+	// closeAwaitingFlush is the state that's transitioned to once both
```

Comment:
Could use some more detail in the commit body message, message says break up, but afaict this is new implementations of process close?

As mentioned before imo can just retain process close (accept `lnwire.Message`, or `lnwire.ChanCloser` (new interface similar to `LinkUpdater`)), which reduces the awareness of the peer dispatch code until we get some sort of message router sub-system going.

Reply:
It's not a new implementation, really. However, I did change the return type substantially to more tightly circumscribe what comes out of each stage.

We can rebundle ReceiveShutdown and ReceiveClosingSigned if we want and I did try that but there's two downsides:

First, there's no way to escape needing to call `BeginNegotiation` when the channel finishes flushing. We used to be able to do this synchronously between the stages because we could assume a clean channel state since we made that a precondition of kicking off this process. We can't anymore.

Second, if we rebundle them, it means that we can't quite as tightly bound the return type as we do here. We instead would need to return a pair of `Option[Shutdown]` and `Option[ClosingSigned]`. If we were to go as far as we did before and inject those types into the larger `[]lnwire.Message` type, then we would need to scrutinize those values in order to insert them into their proper lifecycle hooks (in the case of `Shutdown` it needs to be queued `OnCommitOnce(Outgoing)`).

I did actually write this code out but I removed it because it doesn't actually fix your concern around making this process opaque to the Brontide, and it additionally costs us precision on the actual state machine traversal functions. If you want I can add a commit to the end of this PR demoing what I tried for this and you can take a look at it. If you still prefer it I can pull that work into the main body of this PR in the relevant places, however, I doubt once you see it you'll want to go that route.

Reply:
Ok theres one final problem with rebundling that I believe is fatal. We can't escape inspecting the close message type because we need to decide whether or not to register the flush hook only if the message type is a shutdown. If we receive a valid `closing_signed` then we are already flushed, and shouldn't be flushing again.

Reply:
Ok theres one final problem with rebundling that I believe is fatal. We can't escape inspecting the close message type because we need to decide whether or not to register the flush hook only if the message type is a shutdown. If we receive a valid closing_signed then we are already flushed, and shouldn't be flushing again.

Reply:
See [this](https://github.com/lightningnetwork/lnd/pull/8167/files#r1427456093) for why I think we definitively cannot rebundle.